src/negbinfit.jl

```diff
@@ -0,0 +1,81 @@
+function mleForθ(y::AbstractVector, μ::AbstractVector, wts::AbstractVector;
+                 maxIter = 30, convTol = 1.e-6)
+    function firstDeriv(θ::Real)
+        tmp = (y+θ)./(μ+θ) + log.(μ+θ) - 1 - log.(θ) - digamma.(θ+y) + digamma(θ)
+        unitWeights ? sum(tmp) : sum(wts .* tmp)
+    end
+    function secondDeriv(θ::Real)
+        tmp = -(y+θ)./(μ+θ).^2 + 2 ./ (μ+θ) - 1/θ - trigamma.(θ+y) + trigamma(θ)
+        unitWeights ? sum(tmp) : sum(wts .* tmp)
+    end
+
+    unitWeights = length(wts) == 0
+    n = unitWeights ? length(y) : sum(wts)
+    θ = n / (unitWeights ? sum((y ./ μ - 1).^2) : sum(wts .* (y ./ μ - 1).^2))
+    δ, converged = 1, false
+    for t = 1:maxIter
+        θ = abs(θ)
+        δ = firstDeriv(θ) / secondDeriv(θ)
+        if (abs(δ) <= convTol)
+            converged = true
+            break
+        end
+        θ = θ - δ
+    end
+    converged || throw(ConvergenceException(maxIter))
+    θ
+end
+
+function negbin(F, D, args...; kwargs...)
+    # if initialθ was specified, then use it to train the initial model, remove the argument from fitargs;
+    # otherwise, the initial model is trained with Poisson regression
+    new_kwargs = Dict(kwargs)
+    initialθset = haskey(new_kwargs, :initialθ)
+    maxIter = haskey(new_kwargs, :maxIter) ? new_kwargs[:maxIter] : 30
```

Comment:
Why not use the standard keyword argument syntax? It doesn't prevent you from passing these arguments (manually) do `glm`.